apps/examples/ftpc/ftpc_main.c

```diff
@@ -0,0 +1,464 @@
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdlib.h>
+#include <string.h>
+#include "ftpc.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define FTPC_MAX_ARGUMENTS 4
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+struct cmdmap_s
+{
+ 	const char *cmd;	/* Name of the command */
+ 	cmd_t handler;		/* Function that handles the command */
+ 	uint8_t minargs;	/* Minimum number of arguments (including command) */
+	uint8_t maxargs;	/* Maximum number of arguments (including command) */
+	const char *usage;	/* Usage instructions for 'help' command */
+};
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const char g_delim[] = " \t\n";
+
+static int cmd_lhelp(SESSION handle, int argc, char **argv);
+static int cmd_lunrecognized(SESSION handle, int argc, char **argv);
+
+static pthread_t pth = -1;
+static bool async_cmd_flag;
+
+static const struct cmdmap_s g_cmdmap[] =
+{
+	{ "cd",       cmd_rchdir,  2, 2, "<directory>" },
+	{ "chmod",    cmd_rchmod,  3, 3, "<permissions> <path>" },
+	{ "get",      cmd_rget,    2, 4, "[-a|b] <rname> [<lname>]" },
+	{ "help",     cmd_lhelp,   1, 2, "" },
+	{ "idle",     cmd_ridle,   1, 2, "[<idletime>]" },
+	{ "login",    cmd_rlogin,  2, 3, "<uname> [<password>]" },
+	{ "ls",       cmd_rls,     1, 2, "[<dirpath>]" },
+	{ "quit",     cmd_rquit,   1, 1, "" },

```

Comment:
Except this, all are listed by Arphabet order. Let's move this line between p and r.